src/test/java/seedu/clinic/logic/commands/FindCommandTest.java

```diff
@@ -0,0 +1,200 @@
+import static seedu.clinic.commons.core.Messages.MESSAGE_SUPPLIERS_LISTED_OVERVIEW;
+import static seedu.clinic.commons.core.Messages.MESSAGE_WAREHOUSE_LISTED_OVERVIEW;
+import static seedu.clinic.logic.commands.CommandTestUtil.assertCommandSuccess;
+import static seedu.clinic.testutil.TypicalSupplier.CARL;
+import static seedu.clinic.testutil.TypicalSupplier.ELLE;
+import static seedu.clinic.testutil.TypicalSupplier.FIONA;
+import static seedu.clinic.testutil.TypicalSupplier.getTypicalClinic;
+import static seedu.clinic.testutil.TypicalWarehouse.ALICE;
+import static seedu.clinic.testutil.TypicalWarehouse.BENSON;
+import static seedu.clinic.testutil.TypicalWarehouse.getTypicalWarehouseOnlyClinic;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.clinic.model.Model;
+import seedu.clinic.model.ModelManager;
+import seedu.clinic.model.UserPrefs;
+import seedu.clinic.model.supplier.SupplierPredicate;
+import seedu.clinic.model.warehouse.WarehousePredicate;
+
+/**
+ * Contains integration tests (interaction with the Model) for {@code FindCommand}.
+ */
+public class FindCommandTest {
+    private Model modelForSupplier = new ModelManager(getTypicalClinic(), new UserPrefs());
+    private Model expectedModelForSupplier = new ModelManager(getTypicalClinic(), new UserPrefs());
+    private Model modelForWarehouse = new ModelManager(getTypicalWarehouseOnlyClinic(), new UserPrefs());
+    private Model expectedModelForWarehouse = new ModelManager(getTypicalWarehouseOnlyClinic(), new UserPrefs());
+
+    @Test
+    public void equals() {
+        // Supplier
+        SupplierPredicate firstSupplierPredicate = new SupplierPredicate(Collections.singletonList("first supplier"),
+                Collections.singletonList("first supplier"),
+                Collections.singletonList("first supplier"));
+        SupplierPredicate secondSupplierPredicate = new SupplierPredicate(Collections.singletonList("second supplier"),
+                Collections.singletonList("second supplier"),
+                Collections.singletonList("second supplier"));
+        // Warehouse
+        WarehousePredicate firstWarehousePredicate =
+                new WarehousePredicate(Collections.singletonList("first warehouse"),
+                Collections.singletonList("first warehouse"),
+                Collections.singletonList("first warehouse"));
+        WarehousePredicate secondWarehousePredicate =
+                new WarehousePredicate(Collections.singletonList("second warehouse"),
+                Collections.singletonList("second warehouse"),
+                Collections.singletonList("second warehouse"));
+
+
+        FindCommand firstFindCommand = new FindCommand(firstSupplierPredicate);
+        FindCommand secondFindCommand = new FindCommand(secondSupplierPredicate);
+        FindCommand thirdFindCommand = new FindCommand(firstWarehousePredicate);
```

Comment:
Can consider renaming the function name to contain Supplier and Warehouse for better readability